x/staking/src/types/tx.rs

```diff
@@ -219,28 +222,45 @@ impl Description {
     /// returned if the resulting description contains an invalid length.
     pub fn create_updated_description(
         &self,
-        other: &DescriptionUpdate,
+        other: &Description,
     ) -> Result<Description, anyhow::Error> {
         let mut description = self.clone();
-        if let Some(moniker) = &other.moniker {
-            description.moniker.clone_from(moniker);
+
+        if &other.moniker == DO_NOT_MODIFY_DESCRIPTION {
+            description.moniker = self.moniker.clone();
+        } else {
+            description.moniker = other.moniker.clone();
         }
-        if let Some(identity) = &other.identity {
-            description.identity.clone_from(identity);
+
+        if &other.identity == DO_NOT_MODIFY_DESCRIPTION {
```

Comment:
So, I still could modify this field and this won't result in any error? What the point of this code?

Reply:
What do you mean?

Reply:
What real difference between  `Option` and `DO_NOT_MODIFY_DESCRIPTION`?

Reply:
Agreed to use domain type to isolate [do-not-modify] part